### PR TITLE
Junit5

### PR DESCRIPTION
**Pull request by johnmay**
WIP - but almost done.
Some things to double check as per commit.

- [x] Remove old Assert; usages
- [x] Remove old @Test
- [x] MayGen Paramerterized test
- [x] Remove vintage-engine
- [x] SlowTest tags
- [x] Check the testModelBuilder3D_reserpine() failure
- [x] Verify test counts

### CODE REVIEW

**Comment by johnmay** [comment]
Doing a before/after on the log - some tests have gone missing (probably missing annotations):

```
[john@sentinel:cdk]% diff main.log.norm after.log.norm
55c55
< [INFO] Tests run: 50, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.MappingTest
---
> [INFO] Tests run: 49, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.MappingTest
66c66
< [INFO] Tests run: 4425, Failures: 0, Errors: 0, Skipped: 0
---
> [INFO] Tests run: 4424, Failures: 0, Errors: 0, Skipped: 0
97c97
< [INFO] Tests run: 50, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.silent.MappingTest
---
> [INFO] Tests run: 49, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.silent.MappingTest
107c107
< [INFO] Tests run: 4192, Failures: 0, Errors: 0, Skipped: 0
---
> [INFO] Tests run: 4191, Failures: 0, Errors: 0, Skipped: 0
124c124
< [INFO] Tests run: 50, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.debug.DebugMappingTest
---
> [INFO] Tests run: 49, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.debug.DebugMappingTest
161c161
< [INFO] Tests run: 3980, Failures: 0, Errors: 0, Skipped: 0
---
> [INFO] Tests run: 3979, Failures: 0, Errors: 0, Skipped: 0
191,197d190
< [INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLStackTest
< [INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JumboTest
< [INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JChemPaintTest
< [INFO] Tests run: 25, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLFragmentsTest
< [INFO] Tests run: 30, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CML23FragmentsTest
< [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CML25FragmentsTest
< [INFO] Tests run: 17, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.Jumbo46CMLFragmentsTest
224c217
< [INFO] Tests run: 494, Failures: 0, Errors: 0, Skipped: 0
---
> [INFO] Tests run: 406, Failures: 0, Errors: 0, Skipped: 0
```

**Comment by johnmay** [comment]
```
un: 18, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.HINReaderTe	[INFO] Tests run: 18, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.HINReaderTe
[INFO] Tests run: 25, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLFrag	[INFO] Tests run: 25, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLFrag
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CML25Fra	[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CML25Fra
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLStack	[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLStack
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JmolTest	[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JmolTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JChemPai	[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JChemPai
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JumboTes	[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JumboTes
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLModul	[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLModul
[INFO] Tests run: 17, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.Jumbo46	[INFO] Tests run: 17, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.Jumbo46
[INFO] Tests run: 30, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CML23Fr	[INFO] Tests run: 30, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CML23Fr
[INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLStack   <
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JumboTes   <
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JChemPai   <
[INFO] Tests run: 25, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLFrag   <
[INFO] Tests run: 30, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CML23Fr   <
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CML25Fra   <
[INFO] Tests run: 17, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.Jumbo46   <
```

**Comment by johnmay** [comment]
Fixed the missing annotations and the CML ones were previously run twice.

**Comment by egonw** [comment]
I vaguely remember thinking about running the CML tests more than once, with different XML parsing libraries. (But would not worry about it now anymore)

**Comment by johnmay** [comment]
``
mvn clean verify | grep -F 'Tests run' | sed 's/, Time elapsed: [^ ]* s//' > before.log
``

**Comment by johnmay** [comment]
BINGO! Final "vintage" engine remove flagged it up

``
 */
@RunWith(value = Suite.class)
@SuiteClasses(value = {CMLStackTest.class, JumboTest.class, JChemPaintTest.class, CMLFragmentsTest.class,
        CML23FragmentsTest.class, CML25FragmentsTest.class, Jumbo46CMLFragmentsTest.class})
public class CMLIOTests {
}
``

**Comment by johnmay** [comment]
testModelBuilder3D_reserpine() (SlowTest) is failing on master so once those checks complete I think this is good to go.

File: base/test/src/main/java/org/openscience/cdk/test/interfaces/AbstractMappingTest.java [hunk 1]
```diff
@@ -22,6 +23,7 @@ public abstract class AbstractMappingTest extends AbstractChemObjectTest {
     /**
      * Method to test whether the class complies with RFC #9.
      */
+    @Test
```
Comment (egonw) [inline base/test/src/main/java/org/openscience/cdk/test/interfaces/AbstractMappingTest.java:26]: Ah, I need to check, but this annotation is wrong. These tests should not be run separately, but are inherited by subclasses. I'm adding while going through commit one by one, so the final patch may not actually have the added @Test here.

Comment (egonw) [reply base/test/src/main/java/org/openscience/cdk/test/interfaces/AbstractMappingTest.java:26]: @johnmay, are you sure that this class is not now run as a separate class too?

Comment (johnmay) [reply base/test/src/main/java/org/openscience/cdk/test/interfaces/AbstractMappingTest.java:26]: It think it was abstract, check the log files with the verify/sed file

**Comment by johnmay** [comment]
```
[john@sentinel:cdk]% diff main.log.norm after2.log
55c55
< [INFO] Tests run: 50, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.MappingTest
---
> [INFO] Tests run: 54, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.MappingTest
66c66
< [INFO] Tests run: 4425, Failures: 0, Errors: 0, Skipped: 0
---
> [INFO] Tests run: 4429, Failures: 0, Errors: 0, Skipped: 0
97c97
< [INFO] Tests run: 50, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.silent.MappingTest
---
> [INFO] Tests run: 54, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.silent.MappingTest
107c107
< [INFO] Tests run: 4192, Failures: 0, Errors: 0, Skipped: 0
---
> [INFO] Tests run: 4196, Failures: 0, Errors: 0, Skipped: 0
124c124
< [INFO] Tests run: 50, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.debug.DebugMappingTest
---
> [INFO] Tests run: 54, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.debug.DebugMappingTest
161c161
< [INFO] Tests run: 3980, Failures: 0, Errors: 0, Skipped: 0
---
> [INFO] Tests run: 3984, Failures: 0, Errors: 0, Skipped: 0
191,197d190
< [INFO] Tests run: 8, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLStackTest
< [INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JumboTest
< [INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.JChemPaintTest
< [INFO] Tests run: 25, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CMLFragmentsTest
< [INFO] Tests run: 30, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CML23FragmentsTest
< [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.CML25FragmentsTest
< [INFO] Tests run: 17, Failures: 0, Errors: 0, Skipped: 0 - in org.openscience.cdk.io.cml.Jumbo46CMLFragmentsTest
224c217
< [INFO] Tests run: 494, Failures: 0, Errors: 0, Skipped: 0
---
> [INFO] Tests run: 406, Failures: 0, Errors: 0, Skipped: 0
```

**Pull request merged**